### PR TITLE
Release files

### PR DESCRIPTION
**Pull request by asutula**
Adding files that will be used to automate release of artifacts. Opening this PR so we can discuss how to best integrate into the build process.

### CODE REVIEW

**Comment by asutula** [comment]
Article about pushing to Cocoapods trunk from CircleCI https://fuller.li/posts/automated-cocoapods-releases-with-ci/

File: release-material/cocoapod/TextileCore.podspec [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  spec.license      = "MIT"
+  spec.author       = { "textile.io" => "contact@textile.io" }
+  spec.platform     = :ios, "7.0"
+  spec.source       = spec.source = { :http => 'https://github.com/textileio/go-textile/releases/download/v<version>/go-textile_v<version>_ios-framework.tar.gz' }
+  spec.source_files = "protos"
+  spec.vendored_frameworks = 'Mobile.framework'
+  spec.requires_arc = false
+  spec.dependency "Protobuf", "~> 3.7"
+  spec.pod_target_xcconfig = { 'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1', 'OTHER_LDFLAGS[arch=i386]' => '-Wl,-read_only_relocs,suppress' }
+end
```
Comment (asutula) [inline release-material/cocoapod/TextileCore.podspec]: After github artifacts are published, `<version>` needs to be replaced and we need to run `pod trunk push`.

File: release-material/maven-mobile/io/textile/mobile/VERSION/mobile-VERSION.aar.placeholder [hunk 2]
```diff
@@ -0,0 +1 @@
+This is a placeholder for the real .aar
```
Comment (asutula) [inline release-material/maven-mobile/io/textile/mobile/VERSION/mobile-VERSION.aar.placeholder]: The aar needs to be renamed and copied to this locaion

File: release-material/maven-mobile/io/textile/mobile/VERSION/mobile-VERSION.pom [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <name>Textile</name>
+      <email>contact@textile.io</email>
+    </developer>
+  </developers>
+  <scm>
+    <connection>https://github.com/textileio/go-textile.git</connection>
+    <developerConnection>https://github.com/textileio/go-textile.git</developerConnection>
+    <url>https://github.com/textileio/go-textile</url>
+  </scm>
+</project>
```
Comment (asutula) [inline release-material/maven-mobile/io/textile/mobile/VERSION/mobile-VERSION.pom]: `<VERSION>` needs to be updated in there and in the path to here. Then this whole directory tree will be zipped up and uploaded to Bintray. Will post a link to their API docs.

File: release-material/maven-pb/TextileExample/pb/build.gradle [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    publishedGroupId = 'io.textile'
+    libraryName = 'pb'
+    artifact = 'pb'
+
+    libraryDescription = 'Generated protobuf classes for Textile'
+
+    siteUrl = 'https://github.com/textileio/go-textile'
+    gitUrl = 'https://github.com/textileio/go-textile.git'
+
+    libraryVersion = '<version>'
```
Comment (asutula) [inline release-material/maven-pb/TextileExample/pb/build.gradle]: This version needs to be updated

File: release-material/maven-pb/TextileExample/pb/build.gradle [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    archives javadocJar
+    archives sourcesJar
+}
+
+Properties properties = new Properties()
+properties.load(project.rootProject.file('local.properties').newDataInputStream())
+
+bintray {
+    user = properties.getProperty("bintray.user")
+    key = properties.getProperty("bintray.apikey")
```
Comment (asutula) [inline release-material/maven-pb/TextileExample/pb/build.gradle]: Need to figure out how to read these from env vars

File: release-material/maven-pb/TextileExample/pb/src/test/java/io/textile/pb/ExampleUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package io.textile.pb;
```
Comment (asutula) [inline release-material/maven-pb/TextileExample/pb/src/test/java/io/textile/pb/ExampleUnitTest.java]: Basically all the generated java classes need to be copied into here to replace these

**Comment by sanderpick** [comment]
Nice, I'm out of my wheelhouse here. Is `TextileExample` gonna stay, or is it really just an example? 

What do you need from me here?

**Comment by asutula** [comment]
TextileExample is a project that needs to be build with the generated pb classes, so it will stay. 

What I need from you is:
* Your opinion about the best way organize all this not to clutter up the repo
* Your shell scripting expertise to do things like string replacement in files

**Comment by asutula** [comment]
Maybe a good path forward is I can start making changes to the `config.yml` and get feedback/help from you there.

**Comment by sanderpick** [comment]
Sounds good. My shell foo == crawling stack overflow. Happy to help anywhere. Maybe we could call the top-level folder "release" vs "release-material"?

**Comment by asutula** [comment]
This is a mess

**Pull request closed without merging**